kafka-rest/src/main/java/io/confluent/kafkarest/resources/v3/AclsResource.java

```diff
@@ -158,25 +159,13 @@ public void createAcl(
       @Suspended AsyncResponse asyncResponse,
       @PathParam("clusterId") String clusterId,
       @Valid CreateAclRequest request) {
-    if (request.getResourceType() == Acl.ResourceType.ANY
-        || request.getResourceType() == Acl.ResourceType.UNKNOWN) {
-      throw new BadRequestException("resource_type cannot be ANY");
-    }
-    if (request.getPatternType() == Acl.PatternType.ANY
-        || request.getPatternType() == Acl.PatternType.MATCH
-        || request.getPatternType() == Acl.PatternType.UNKNOWN) {
-      throw new BadRequestException(
-          String.format("pattern_type cannot be %s", request.getPatternType()));
-    }
-    if (request.getOperation() == Acl.Operation.ANY
-        || request.getOperation() == Acl.Operation.UNKNOWN) {
-      throw new BadRequestException("operation cannot be ANY");
-    }
-    if (request.getPermission() == Acl.Permission.ANY
-        || request.getPermission() == Acl.Permission.UNKNOWN) {
-      throw new BadRequestException("permission cannot be ANY");
+
+    if (request == null) {
+      throw Errors.invalidPayloadException("Null input provided. Data is required.");
     }
```

Comment:
nit: I wonder if the `@Valid` annotation doesn't preclude this check from happening. Below I've suggested adding a test that checks that case.

Reply:
I put this in because I manually tested that I got 500s without it, so it is needed.